src/v2/coins/abstractUtxoCoin.ts

```diff
@@ -1081,48 +1081,28 @@ class AbstractUtxoCoin extends BaseCoin {
         }
       }
 
-      if (hasSigScript) {
-        // p2sh or p2shP2wsh
-        const nonEmptySignatures = sigScript.signatures.filter((sig) => !_.isEmpty(sig));
-        const validSignatures = nonEmptySignatures.map((sig, sigIndex) => {
-          if (_.isEmpty(sig)) {
-            return false;
-          }
-
-          const verificationParams = {
-            signatureIndex: sigIndex
-          };
-
-          if (hasWitnessScript) {
-            // p2shP2wsh
-            const parentTxId = Buffer.from(input.hash).reverse().toString('hex');
-            const inputId = `${parentTxId}:${input.index}`;
-            const amount = unspentValues[inputId];
-
-            try {
-              return this.verifySignature(transaction, idx, amount, verificationParams);
-            } catch (e) {
-              return false;
-            }
-          }
+      const nonEmptySignatures = parsedSigScript.signatures.filter((sig) => !_.isEmpty(sig));
+      const validSignatures = nonEmptySignatures.map((sig, sigIndex) => {
+        if (_.isEmpty(sig)) {
+          return false;
+        }
 
-          // p2sh
-          try {
-            return this.verifySignature(transaction, idx, undefined, verificationParams);
-          } catch (e) {
-            return false;
-          }
-        });
+        const verificationParams = {
```

Comment:
nit: unnecessary var

Reply:
I like that better. Fixed and amended the commit. Please re-review :)